src/LMS1xx.cpp

```diff
@@ -117,9 +118,24 @@ status_t LMS1xx::queryStatus() {
 
 void LMS1xx::login() {
 	char buf[100];
+	int result;
 	sprintf(buf, "%c%s%c", 0x02, "sMN SetAccessMode 03 F4724744", 0x03);
+	
+	fd_set readset;
+	struct timeval timeout;
+	timeout.tv_sec = 1;
+	timeout.tv_usec = 0;
 
-	write(sockDesc, buf, strlen(buf));
+	do { //loop until data is available to read
+	  ros::Duration(1.0).sleep();
```

Comment:
If this is what's creating the `ros.h` dependency, you can pause for 1 second in other ways, for example simply `select(0, NULL, NULL, NULL, &timeout);`.